File: src/main/java/net/reflact/common/network/PacketRegistry.java
package net.reflact.common.network;

import net.reflact.common.network.packet.*;
import java.util.Map;
import java.util.HashMap;

public class PacketRegistry {
    private static final Map<String, Class<? extends ReflactPacket>> REGISTRY = new HashMap<>();

    static {
        REGISTRY.put("cast_spell", CastSpellPacket.class);
        REGISTRY.put("mana_update", ManaUpdatePacket.class);
        REGISTRY.put("sync_item", S2CSyncItemPacket.class);
    }

    public static Class<? extends ReflactPacket> get(String id) {
        return REGISTRY.get(id);
    }
    
    public static String getId(Class<? extends ReflactPacket> clazz) {
        for (var entry : REGISTRY.entrySet()) {
            if (entry.getValue().equals(clazz)) return entry.getKey();
        }
        return null;
    }
}

File: bin/main/net/reflact/common/network/PacketRegistry.kt
package net.reflact.common.network

import net.reflact.common.network.packet.*

object PacketRegistry {
    private val REGISTRY = HashMap<String, Class<out ReflactPacket>>()

    init {
        REGISTRY["cast_spell"] = CastSpellPacket::class.java
        REGISTRY["mana_update"] = ManaUpdatePacket::class.java
        REGISTRY["sync_item"] = S2CSyncItemPacket::class.java
        REGISTRY["cast_slot"] = CastSlotPacket::class.java
        REGISTRY["map_data"] = MapDataPacket::class.java
    }

    fun get(id: String): Class<out ReflactPacket>? {
        return REGISTRY[id]
    }

    fun getId(clazz: Class<out ReflactPacket>): String? {
        for ((key, value) in REGISTRY) {
            if (value == clazz) return key
        }
        return null
    }
}
